mitodl/mit-open
# search facet accessibility fixes

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5421
Closes https://github.com/mitodl/hq/issues/5429

### Description (What does it do?)
This PR takes advantage of the `facet-transitioning` class added in https://github.com/mitodl/course-search-utils/pull/146, allowing us to set `visibility: hidden` on `facet-visible` items while they are not expanded or transitioning. The default color of the expand / collapse caret on facets was also changed to `silverGrayDark` in accordance with https://github.com/mitodl/hq/issues/5429.

### How can this be tested?
First, we will need to get a new version of `course-search-utils`
 - Clone `course-search-utils` on the `cg/add-facet-transitioning-class` branch: https://github.com/mitodl/course-search-utils
 - Once you are on the correct branch, run `yarn install` and then `yarn typescript` to generate the library
 - Back in `mit-learn`, overwrite the contents of the `node_modules/@mitodl/course-search-utils/dist` folder with the contents of `course-search-utils/dist` that we just built
 - Spin up `mit-learn` on the `cg/facet-accessibility` branch
 - Make sure you have some data backpopulated in your database and a search index built
 - Browse to the search page at https://localhost:8062/search
 - Collapse one of the search facets on the left hand side, then right click on the facet and click Inspect
 - Look at the `facet-visible` items within the collapsed facet and ensure they have the `visibility: hidden` rule applied to them
 - Expand the facet while watching the CSS rules on one of the `facet-visible` items
 - Verify that the `visibility: hidden` rule is removed

### Checklist:
- [ ] Merge and release https://github.com/mitodl/course-search-utils/pull/146
- [ ] Update the version of `course-search-utils` in `package.json` with the newly released version in this PR before merging it
